tox.ini

```diff
@@ -1,8 +1,8 @@
 [tox]
 envlist =
-    {py27,py34,py35,py36}-django-18
-    {py27,py34,py35,py36}-django-111
-    {py34,py35,py36}-django-20
+    {py27,py34,py35,py36}-django-18-{slugify,noslugify}
```

Comment:
do you mind just explaining how this works for my own edification? I see that it is testing with awesome-slugify and without it but not sure how

Reply:
Sure! it was an educational exercise for myself too!

The line {py27,py34,py35,py36}-django-18-{slugify,noslugify}` creates 8 environments:
```
py27-django-18-slugify
py27-django-18-noslugify
...
py36-django-18-noslugify
```

line 15 (`slugify: -r{toxinidir}/requirements.txt`) tells tox that if the env contains 'slugify', it should install `awesome-slugify`.

Reply:
I realised after the fact that tox must actually be checking for the the presence of `slugify` not in the raw string name of the env, but in something like `envname.split('-')`, because otherwise _every_ env would install `awesome-slugify`, as "noslugify" obviously contains "slugify". However, I checked the output of tox and it seems that this has worked as intended.